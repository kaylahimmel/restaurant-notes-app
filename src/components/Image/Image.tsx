import NextImage from 'next/image'

export interface ImageProps {
  src: string
  alt: string
  width?: number // default is 500px
  height?: number // default is 500px
  overrideSrc?: string
}

export default function Image({
  src,
  alt,
  width,
  height,
  overrideSrc,
}: ImageProps) {
  return (
    <NextImage
      src={src}
      alt={alt}
      width={width}
      height={height}
      overrideSrc={overrideSrc}
    />
  )
}
