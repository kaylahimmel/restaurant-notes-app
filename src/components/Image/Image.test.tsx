import React from 'react'
import { render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import Image from './Image'

describe('Image component', () => {
  it('renders an image from a local source', () => {
    const { getByAltText } = render(
      <Image
        src="../../public/profile.svg"
        alt="profile image"
        width={200}
        height={200}
      />
    )
    const imgElement = getByAltText('profile image')
    expect(imgElement).toBeInTheDocument()
    expect(imgElement).toHaveAttribute('src', '../../public/profile.svg')
  })

  it('renders an image from a remote URL', async () => {
    const { getByAltText } = render(
      <Image
        src="http://source.unsplash.com/photos/green-leaves-with-white-background-hVlf2TDEXV8"
        alt="monstera leaf"
        width={400}
        height={600}
        overrideSrc="http://source.unsplash.com/photos/green-leaves-with-white-background-hVlf2TDEXV8"
      />
    )
    const imgElement = getByAltText('monstera leaf')

    await waitFor(() => {
      expect(imgElement).toBeInTheDocument()
      expect(imgElement).toHaveAttribute(
        'src',
        'http://source.unsplash.com/photos/green-leaves-with-white-background-hVlf2TDEXV8'
      )
    })
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <Image
        src="https://placehold.co/150x350"
        alt="placeholder"
        width={150}
        height={350}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
