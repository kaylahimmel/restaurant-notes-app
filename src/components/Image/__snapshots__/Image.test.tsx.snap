// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component matches snapshot 1`] = `
<DocumentFragment>
  <img
    alt="placeholder"
    data-nimg="1"
    decoding="async"
    height="350"
    loading="lazy"
    src="/_next/image?url=https%3A%2F%2Fplacehold.co%2F150x350&w=384&q=75"
    srcset="/_next/image?url=https%3A%2F%2Fplacehold.co%2F150x350&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fplacehold.co%2F150x350&w=384&q=75 2x"
    style="color: transparent;"
    width="150"
  />
</DocumentFragment>
`;
